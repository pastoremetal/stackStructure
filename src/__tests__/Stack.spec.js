const Stack = require("../Stack");

describe('5 non sequential integers and floats with repeated values', () => {
    const numbers = [43, 23, 9, 7, 2, 8, 65, 0.2, 1, 1, 59.3, 0.1, 0.1];
    for (const i of numbers) {
        Stack.addValue(i);
    }

    it('add more 5 non sequential integers and should match snapshot', () => {
        expect(Stack.viewStack()).toMatchSnapshot();
    });

    it('should get value on top of stack', () => {
        expect(Stack.viewTopValue()).toBe(numbers[numbers.length -1]);
    });

    it('should remove value from top of stack', () => {
        Stack.removeTopValue();
        expect(Stack.viewStack()).toMatchSnapshot();
    });

    it('should get smaller value of stack', () => {
        expect(Stack.viewSmallerValue()).toBe(0.1);
    });
});