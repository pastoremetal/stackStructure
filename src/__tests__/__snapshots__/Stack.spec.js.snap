// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`5 non sequential integers and floats with repeated values add more 5 non sequential integers and should match snapshot 1`] = `
Array [
  43,
  23,
  9,
  7,
  2,
  8,
  65,
  0.2,
  1,
  1,
  59.3,
  0.1,
  0.1,
]
`;

exports[`5 non sequential integers and floats with repeated values should remove value from top of stack 1`] = `
Array [
  43,
  23,
  9,
  7,
  2,
  8,
  65,
  0.2,
  1,
  1,
  59.3,
  0.1,
]
`;
